### PR TITLE
Removing glob2 and replacing with glob(,recursive=True) where used.

### PR DESCRIPTION
**Pull request by ryanskeith**
### Description

Removed glob2 from code, requirements and recipes.
- Removed code
- Replaced all places with glob(..., recursive=True) for same functionality.
- glob2.fnmatch.fnmatch references seem to be the simple cases of fnmatch

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
- [x] Add / update necessary tests?
- [x] Add / update outdated documentation?

### CODE REVIEW

File: news/4792-remove-glob2 [hunk 1]
```diff
@@ -0,0 +1,3 @@
+### Enhancements
+
+* Remove glob2 as a dependency.  As of Python 3.5, the '**', operator was available to glob when using recursive=True.  Builtin glob is also much faster.
```
Comment (beeankha) [inline news/4792-remove-glob2:3]:
```suggestion
* Remove `glob2` as a dependency.  As of Python 3.5, the '**', operator was available to glob when using `recursive=True`.  Builtin glob is also much faster. (#5005)
```

File: conda_build/build.py [hunk 2]
```diff
@@ -1205,10 +1204,7 @@ def get_files_with_prefix(m, replacements, files_in, prefix):
                 files=[
                     f
                     for f in files
-                    if any(
-                        glob2.fnmatch.fnmatch(f, r)
-                        for r in replacement["glob_patterns"]
-                    )
+                    if any(fnmatch.fnmatch(f, r) for r in replacement["glob_patterns"])
```
Comment (kenodegard) [inline conda_build/build.py:1207]: Let's switch out these single char vars:
```
files=[
    file
    for file in files
    if any(fnmatch.fnmatch(file, pattern) for pattern in replacement["glob_patterns"])
]
```

File: conda_build/utils.py [hunk 3]
```diff
@@ -63,7 +63,10 @@
 
 import urllib.parse as urlparse
 import urllib.request as urllib
-from glob import glob as glob_glob
+
+# NOQA because it is not used in this file.
+from contextlib import ExitStack  # NOQA
```
Comment (kenodegard) [inline conda_build/utils.py:68]:
```suggestion
from contextlib import ExitStack  # noqa: F401
```

**Comment by kenodegard** [comment]
pre-commit.ci autofix

**Review by kenodegard — APPROVED** [review]

**Comment by ryanskeith** [comment]
This is addressing #4792.

**Comment by dholth** [comment]
How does this compare to https://docs.python.org/3/library/pathlib.html#pathlib.Path.glob ?

**Comment by ryanskeith** [comment]
> How does this compare to https://docs.python.org/3/library/pathlib.html#pathlib.Path.glob ?

In general @dholth, or is there a specific instance?

**Comment by kenodegard** [comment]
@dholth comparing `glob2`, `glob`, and `pathlib` for `**` globbing, `pathlib` only matches dirs (including the root dir) while `glob2`/`glob` matches any path:

```bash
$ ls -lahR
total 0
drwxr-xr-x@  4 kodegard  staff   128B Sep 25 11:29 .
drwxr-xr-x  77 kodegard  staff   2.4K Sep 25 11:29 ..
-rw-r--r--@  1 kodegard  staff     0B Sep 25 11:29 file
drwxr-xr-x@  4 kodegard  staff   128B Sep 25 11:29 nested

./nested:
total 0
drwxr-xr-x@ 4 kodegard  staff   128B Sep 25 11:29 .
drwxr-xr-x@ 4 kodegard  staff   128B Sep 25 11:29 ..
-rw-r--r--@ 1 kodegard  staff     0B Sep 25 11:29 file
drwxr-xr-x@ 3 kodegard  staff    96B Sep 25 11:29 nested

./nested/nested:
total 0
drwxr-xr-x@ 3 kodegard  staff    96B Sep 25 11:29 .
drwxr-xr-x@ 4 kodegard  staff   128B Sep 25 11:29 ..
-rw-r--r--@ 1 kodegard  staff     0B Sep 25 11:29 file

$ python - <<HEREDOC
import glob2, glob, pathlib
print(*glob2.glob('**'))
print(*glob.glob('**', recursive=True))
print(*pathlib.Path.cwd().glob('**'))
HEREDOC
file nested nested/file nested/nested nested/nested/file
file nested nested/file nested/nested nested/nested/file
/Users/kodegard/scratch/glob /Users/kodegard/scratch/glob/nested /Users/kodegard/scratch/glob/nested/nested
```

**Pull request merged**